File: spring-data-requery/src/test/java/org/springframework/data/requery/repository/sample/UserRepositoryCustom.java
package org.springframework.data.requery.repository.sample;

import org.springframework.data.requery.domain.sample.User;

/**
 * UserRepositoryCustom
 *
 * @author debop@coupang.com
 * @since 18. 6. 25
 */
public interface UserRepositoryCustom {

    /**
     * Method actually triggering a finder but being overridden.
     */
    void findByOverrridingMethod();

    /**
     * Some custom method to implement.
     */
    void someCustomMethod(User user);
}

File: spring-data-requery/src/test/kotlin/org/springframework/data/requery/repository/sample/UserRepositoryCustom.kt
package org.springframework.data.requery.repository.sample

import org.springframework.data.requery.domain.sample.User

/**
 * org.springframework.data.requery.repository.sample.UserRepositoryCustom
 *
 * @author debop
 */
interface UserRepositoryCustom {

    /**
     * Method actually triggering a finder but being overridden.
     */
    fun findByOverrridingMethod()

    /**
     * Some custom method to implement.
     */
    fun someCustomMethod(user: User)
}